src/proto/media/file/mod.rs

```diff
@@ -280,6 +288,22 @@ impl File for FileHandle {
     fn handle(&mut self) -> &mut FileHandle {
         self
     }
+
+    fn is_regular_file(&self) -> bool {
+        let this = unsafe { self.0.as_mut().unwrap() };
+
+        // get_position fails with EFI_UNSUPPORTED on directories
+        let mut pos = 0;
+        match (this.get_position)(this, &mut pos) {
+            Status::SUCCESS => true,
+            Status::UNSUPPORTED => false,
+            _ => panic!("unexpected response"),
```

Comment:
if this panics, we have to find a better solution in the crate. It is a hard bug then. Not something users should cope with

Reply:
Does the spec describe when an error could occur? If it's something which might happen usually in practice, like spurious read errors, then it might be better to have these functions return a `Result`.

Reply:
The spec  does list one other error case: `EFI_DEVICE_ERROR` if "An attempt was made to get the position from a deleted file". I think that's enough justification for changing the return type to `Result<bool>` here, a la [`try_exists`](https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists).

Reply:
I agree. As this is a low-level API, this is fine. I think, I can relax this in #472 as we have high-level control over many situations.